File: py_validation/pardox/_bindings/decimal.py
def bind(lib, c_void_p, c_char_p, c_int32, c_double, c_longlong, c_size_t, c_int64, c_int16):
    for fn, args, ret in [
        ('pardox_decimal_from_float',  [c_void_p, c_char_p, c_int32],           c_void_p),
        ('pardox_decimal_sum',         [c_void_p, c_char_p],                    c_double),
        ('pardox_decimal_add',         [c_void_p, c_char_p, c_void_p, c_char_p], c_void_p),
        ('pardox_decimal_sub',         [c_void_p, c_char_p, c_void_p, c_char_p], c_void_p),
        ('pardox_decimal_mul_float',   [c_void_p, c_char_p, c_double],           c_void_p),
        ('pardox_decimal_round',       [c_void_p, c_char_p, c_int32],           c_void_p),
        ('pardox_decimal_to_float',    [c_void_p, c_char_p],                    c_void_p),
        ('pardox_decimal_get_scale',   [c_void_p, c_char_p],                    c_int32),
    ]:
        if hasattr(lib, fn):
            getattr(lib, fn).argtypes = args
            getattr(lib, fn).restype  = ret

File: js_validation/pardox/src/_bindings/decimal.js
'use strict';

module.exports = function bindDecimal(_lib) {
    _lib.pardox_decimal_from_float = _lib.raw.func('void * pardox_decimal_from_float(void * mgr, const char * col_name, uint8_t scale)');
    _lib.pardox_decimal_to_float   = _lib.raw.func('void * pardox_decimal_to_float(void * mgr, const char * col_name)');
    _lib.pardox_decimal_add        = _lib.raw.func('void * pardox_decimal_add(void * mgr, const char * col_a, const char * col_b)');
    _lib.pardox_decimal_sub        = _lib.raw.func('void * pardox_decimal_sub(void * mgr, const char * col_a, const char * col_b)');
    _lib.pardox_decimal_mul_float  = _lib.raw.func('void * pardox_decimal_mul_float(void * mgr, const char * col_name, double factor)');
    _lib.pardox_decimal_round      = _lib.raw.func('void * pardox_decimal_round(void * mgr, const char * col_name, uint8_t scale)');
    _lib._raw_pardox_decimal_sum   = _lib.raw.func('double pardox_decimal_sum(void * mgr, const char * col_name)');
    _lib.pardox_decimal_get_scale  = _lib.raw.func('int pardox_decimal_get_scale(void * mgr, const char * col_name)');
};
